test/ConfigDistributor.spec.js

```diff
@@ -24,177 +21,100 @@ describe('A DistributeConfig class', () => {
-        const configDistributor = new ConfigDistributor();
-        return configDistributor.distribute(configPaths, configConsolidatorMock, fileWriterMock, directoryWriterMock, reportMock).then(() => {
-          expect(configConsolidatorMock.consolidate.calledOnceWithExactly(configPaths)).to.be.true;
-        });
-      });
-
-    });
-
-  });
-
-  describe('processForSass static method', () => {
-
-    context('when supplied with a JavaScript object defining data for SASS variables', () => {
-
-      let sassConfigFixture;
-
-      beforeEach(() => {
-        sassConfigFixture = fixtures.sassConfigToProcess.input;
-      });
-
-      it('correctly converts the JavaScript into a string wrapping corresponding SASS variable declarations', () => {
-        const processed = ConfigDistributor.processForSass(sassConfigFixture);
-        expect(processed).to.deep.equal(fixtures.sassConfigToProcess.expected);
-      });
-
-    });
-
-  });
-
-  describe('processForJs static method', () => {
-
-    context('when supplied with a JavaScript object defining data for the JavaScript layer', () => {
-
-      let jsConfigFixture;
-
-      beforeEach(() => {
-        jsConfigFixture = fixtures.jsConfigToProcess.input;
-      });
-
-      it('correctly converts the JavaScript into the appropriate JSON', () => {
-        const processed = ConfigDistributor.processForJs(['breakpoints'], jsConfigFixture);
-        expect(processed).to.deep.equal(JSON.stringify(fixtures.jsConfigToProcess.input));
-      });
-
-    });
-
-  });
-
-  describe('writeDirectory static method', () => {
-
-    context('when there is an error creating the directory', () => {
-
-      it('returns a promise that will be rejected', () => {
-        const invalidDirName = (function () {
-          const parts = new Array(256);
-          parts.fill('a');
-          return parts.join('');
-        }());
-        return expect(
-          ConfigDistributor.writeDirectory(invalidDirName)
-        ).to.be.rejected;
-      });
-
-    });
-
-    context('when there no error', () => {
-
-      let validDirName;
+      let consolidatorMock;
+      let consolidateSpy;
+      let distributor;
+      let expected;
+      let fileSystemMock;
+      let paths;
+      let readFileSpy;
+      let writeFileSpy;
 
       beforeEach(() => {
-        validDirName = path.join(__dirname, '/tmp/');
-        if (fs.existsSync(validDirName)) {
-          fs.rmdirSync(validDirName);
-        }
-        expect(fs.existsSync(validDirName)).to.be.false;
-      });
+          consolidatorMock = {
+            consolidate: () => {
+              return Promise.resolve(standAloneConfigFixture.input);
+            }
+          };
+          consolidateSpy = spy(consolidatorMock, 'consolidate');
+
+        fileSystemMock = {
+          writeFile: () => { return Promise.resolve; },
+          readFile: () => { return Promise.resolve; }
+        };
+        writeFileSpy = spy(fileSystemMock, 'writeFile');
```

Comment:
these look more like [mocks](https://sinonjs.org/releases/v7.2.2/mocks/) than spies as they don't wrap the original, real object. That would have you write expectations beforehand rather than verifying the records, probably.